src/vm/contexts.rs

```diff
@@ -606,7 +606,7 @@ impl <'a,'b> Environment <'a,'b> {
 
             let func = contract.contract_context.lookup_function(tx_name)
                 .ok_or_else(|| { CheckErrors::UndefinedFunction(tx_name.to_string()) })?;
-            if !func.is_public() {
+            if !func.is_public() || (read_only && !func.is_read_only()) {
```

Comment:
Shouldn't this be a "Function is read-only" error?

Reply:
Added error type in 5951759a789563a6ecb1ae3800fc0706944b02ec